### PR TITLE
✨Add support for infinite scroll using amp-infinite-scroll component

### PR DESCRIPTION
**Pull request by bodbdigr**
This PR introduces a new component that adds infinite scroll feature to AMP,  idea is that user can add `amp-inifinite-scroll` component and provide next page URL that would be fetched and injected into current page once user hits visibility area of `amp-infinite-scroll` component and it would repeat until there's no more pages to be served via API (see an example https://jmp.sh/BaLibDa )

### CODE REVIEW

**Comment by kristoferbaxter** [comment]
@bodbdigr – Exciting to see this!

Can you accept the CLA so we can start to review this PR?

**Comment by bodbdigr** [comment]
> @bodbdigr – Exciting to see this!
> 
> Can you accept the CLA so we can start to review this PR?

Thank you @kristoferbaxter, sorry for late reply, CLA is accepted now (I think :))

**Comment by bodbdigr** [comment]
> Thanks for your pull request. It looks like this may be your first contribution to a Google open source project (if not, look below for help). Before we can look at your pull request, you'll need to sign a Contributor License Agreement (CLA).
> 
> 📝 **Please visit https://cla.developers.google.com/ to sign.**
> 
> Once you've signed (or fixed any issues), please reply here with `@googlebot I signed it!` and we'll verify it.
> 
> #### What to do if you already signed the CLA
> ##### Individual signers
> * It's possible we don't have your GitHub username or you're using a different email address on your commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> 
> ##### Corporate signers
> * Your company has a Point of Contact who decides which employees are authorized to participate. Ask your POC to be added to the group of authorized contributors. If you don't know who your Point of Contact is, direct the Google project maintainer to [go/cla#troubleshoot](http://go/cla#troubleshoot) ([Public version](https://opensource.google.com/docs/cla/#troubleshoot)).
> * The email used to register you as an authorized contributor must be the email used for the Git commit. Check [your existing CLA data](https://cla.developers.google.com/clas) and verify that your [email is set on your git commits](https://help.github.com/articles/setting-your-email-in-git/).
> * The email used to register you as an authorized contributor must also be [attached to your GitHub account](https://github.com/settings/emails).
> 
> ℹ️ **Googlers: [Go here](https://goto.google.com/prinfo/https%3A%2F%2Fgithub.com%2Fampproject%2Famphtml%2Fpull%2F24889) for more info**.

 @googlebot I signed it!

File: extensions/amp-infinite-scroll/0.1/test/validator-amp-infinite-scroll.html [hunk 1]
```diff
@@ -0,0 +1,20 @@
+<!doctype html>
```
Comment (honeybadgerdontcare) [inline extensions/amp-infinite-scroll/0.1/test/validator-amp-infinite-scroll.html]: This is missing the appropriate license with 2019 as year.

Should also include comment explaining what this file is testing and also each test case within the document. See [this as an example](https://github.com/ampproject/amphtml/blob/master/extensions/amp-autocomplete/0.1/test/validator-amp-autocomplete.html).

Comment (bodbdigr) [reply extensions/amp-infinite-scroll/0.1/test/validator-amp-infinite-scroll.html]: @honeybadgerdontcare thank you, validation rules should be fixed now

**Comment by wassgha** [comment]
Closing and tracking implementation through #25500 as per offline discussion

**Pull request closed without merging**